src/engines/seq/euclid/euclid.cpp

```diff
@@ -203,9 +214,14 @@ namespace otto::engines {
     ctx.beginPath();
     ctx.fillText(
       util::join_strings(util::view::transform(
+                           util::view::filter(current.notes, [](char note) { return note >= 0; }),
                            // extra safe on the note value to avoid overflow
-                           util::view::filter(current.notes, [](char note) { return note >= 0 && note < 128; }),
-                           [](char note) { return midi::note_name(note); }),
+                           [](char note) {
+                             if (note >= 0)
+                               return midi::note_name(note);
+                             else
+                               return "";
```

Comment:
this is impossible, thats what `filter` is for

Reply:
Well, it *should* be impossible, but if a bug is introduced in filter (as it was the case before), then it will crash, I usually prefer to envision the worst case scenario to ensure resilience ^^